src/platform/datapath_raw_socket.c

```diff
@@ -570,7 +570,7 @@ CxPlatResolveRoute(
     CXPLAT_LIST_ENTRY* Entry = Socket->Datapath->Interfaces.Flink;
     for (; Entry != &Socket->Datapath->Interfaces; Entry = Entry->Flink) {
         CXPLAT_INTERFACE* Interface = CONTAINING_RECORD(Entry, CXPLAT_INTERFACE, Link);
-        if (Interface->IfIndex == IpforwardRow.InterfaceIndex) {
+        if (Interface->ActualIfIndex == IpforwardRow.InterfaceIndex) {
```

Comment:
Are you sure this shouldn't be the normal IfIndex? I think it should.

Reply:
I thought this is actual one (VF) because it is copying mac address in the if block.

Reply:
What does the GetBestRoute2 scope? only in VM or entire system?

Reply:
the scope seems to be in OS (VM), IfIndex should be correct

Reply:
I'm not sure of the details, but this is getting info from TCP/IP which works on top of the `IfIndex`, not the actual one, so I think this should align with TCP/IP's view.